dashboard/app/models/resource.rb

```diff
@@ -48,6 +48,23 @@ def generate_key
     self.key = generate_key_from_name
   end
 
+  # Used for seeding from JSON. Returns the full set of information needed to
+  # uniquely identify this object as well as any other objects it belongs to.
+  # If the attributes of this object alone aren't sufficient, and associated
+  # objects are needed, then data from the seeding_keys of those objects should
+  # be included as well. Ideally should correspond to a unique index for this
+  # model's table. See comments on ScriptSeed.seed_from_json for more context.
+  #
+  # @param [ScriptSeed::SeedContext] _seed_context - contains preloaded data to use when looking up associated objects
+  # @return [Hash<String, String>] all information needed to uniquely identify this object across environments.
+  def seeding_key(_seed_context)
+    # Course version is also needed to identify this object, and can be looked
+    # up from the script/unit which this resource is serialized within. If we
+    # were to serialize resources outside of .script_json, we'd need to include
+    # a key respresenting the course version here.
+    {'resource.key': key}.stringify_keys
```

Comment:
Do we need to also include the `id` (or other unique identifier) of the course version? Resource keys are not guaranteed to be unique between course versions.

Reply:
In theory, we shouldn't need the course version because the key is enough information to identify the resource within the script. In practice, we accomplish this by only populating the seed context with resources which are within the script, so this lookup succeeds: https://github.com/code-dot-org/code-dot-org/blob/4bacf9345d0d54e985dc28c0f8f75e30e06a4738/dashboard/lib/services/script_seed.rb#L327-L328

Reply:
Ahhh makes sense -- thanks!